zipline/assets/synthetic.py

```diff
@@ -82,11 +83,16 @@ def make_simple_equity_info(sids,
     num_assets = len(sids)
     if symbols is None:
         symbols = list(ascii_uppercase[:num_assets])
+
+    if names is None:
+        names = map(lambda x: str(x) + " INC.", symbols)
+
     return pd.DataFrame(
         {
             'symbol': list(symbols),
             'start_date': pd.to_datetime([start_date] * num_assets),
             'end_date': pd.to_datetime([end_date] * num_assets),
+            'asset_name': list(names),
```

Comment:
This branch has failing python3 tests in the case where `symbols` is passed as a generator (actually the result of a call to `map`, which in python2 returns a `list`). The map is already consumed by the call to `list(symbols)`, so `names` is an empty list, which is a different length from the length of the list of symbols.